src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/Services/JsonVectorStore.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Microsoft.Extensions.VectorData;
+using System.Numerics.Tensors;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+
+namespace ChatWithCustomData.Web.Services;
+
+/// <summary>
+/// This IVectorStore implementation is for prototyping only. Do not use this in production.
+/// In production, you must replace this with a real vector store. There are many IVectorStore
+/// implementations available, including ones for standalone vector databases like Qdrant or Milvus,
+/// or for integrating with relational databases such as SQL Server or PostgreSQL.
+/// 
+/// This implementation stores the vector records in large JSON files on disk. It is very inefficient
+/// and is provided only for convenience when prototyping.
+/// </summary>
+public class JsonVectorStore(string basePath) : IVectorStore
```

Comment:
It'd be really nice if this didn't need to be part of the template.

Reply:
I have to resolve all the conversations before CI will let me merge this. I'll unresolve after it merges.

Reply:
> It'd be really nice if this didn't need to be part of the template.

100% agreed. Tracked as https://github.com/dotnet/ai-private-planning/issues/270